payments-core/src/main/java/com/stripe/android/payments/core/authentication/DefaultPaymentAuthenticatorRegistry.kt

```diff
@@ -137,25 +123,40 @@ internal class DefaultPaymentAuthenticatorRegistry @Inject internal constructor(
             threeDs1IntentReturnUrlMap: MutableMap<String, String>,
             publishableKeyProvider: () -> String,
             productUsage: Set<String>,
-            isInstantApp: Boolean
+            isInstantApp: Boolean,
+            includePaymentSheetAuthenticators: Boolean,
         ): PaymentAuthenticatorRegistry {
-            val injectorKey =
-                WeakMapInjectorRegistry.nextKey(requireNotNull(PaymentAuthenticatorRegistry::class.simpleName))
             val component = DaggerAuthenticationComponent.builder()
                 .context(context)
                 .analyticsRequestFactory(paymentAnalyticsRequestFactory)
                 .enableLogging(enableLogging)
                 .workContext(workContext)
                 .uiContext(uiContext)
                 .threeDs1IntentReturnUrlMap(threeDs1IntentReturnUrlMap)
-                .injectorKey(injectorKey)
                 .publishableKeyProvider(publishableKeyProvider)
                 .productUsage(productUsage)
                 .isInstantApp(isInstantApp)
+                .includePaymentSheetAuthenticators(includePaymentSheetAuthenticators)
                 .build()
-            val registry = component.registry
-            registry.authenticationComponent = component
-            return registry
+            return component.registry
+        }
+    }
+}
+
+private fun paymentSheetAuthenticators(
+    includePaymentSheetAuthenticators: Boolean
+): Map<AuthenticatorKey, Authenticator> {
```

Comment:
Can we test this reflection to avoid us breaking it by mistake?

Reply:
We do test it via an existing test in payment sheet (it's how I knew this wouldn't work by my original removal of the pattern).

As for testing it more than that, I'm not sure it would make a lot of sense.

Reply:
Oh right, that’s sufficient then. As long as the full flow that’s using reflection is covered, I’m fine with it!